packages/actor-rdf-resolve-hypermedia-qpf/lib/ActorRdfResolveHypermediaQpf.ts

```diff
@@ -0,0 +1,74 @@
+import {ISearchForms} from "@comunica/actor-rdf-metadata-extract-hydra-controls";
+import {KEY_CONTEXT_SOURCE} from "@comunica/bus-rdf-resolve-quad-pattern";
+import {ActorRdfResolveHypermedia, IActionRdfResolveHypermedia,
+  IActorRdfResolveHypermediaOutput} from "@comunica/bus-rdf-resolve-hypermedia";
+import {ActionContext, IActorArgs, IActorTest} from "@comunica/core";
+
+/**
+ * A comunica QPF RDF Resolve Quad Pattern Actor.
+ */
+export class ActorRdfResolveHypermediaQpf extends ActorRdfResolveHypermedia
+implements IActorRdfResolveHypermediaQpfArgs {
+
+  public readonly subjectUri: string;
+  public readonly predicateUri: string;
+  public readonly objectUri: string;
+  public readonly graphUri?: string;
+
+  constructor(args: IActorRdfResolveHypermediaQpfArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfResolveHypermedia): Promise<IActorTest> {
+    if (!(action.context.get(KEY_CONTEXT_SOURCE).type === "hypermedia")) {
+      throw new Error(this.name
+        + ' requires a single source with a Hypermedia \'hypermedia\' entrypoint to be present in the context.');
+    }
+
+    if (!action.metadata || !action.metadata.searchForms) {
+      throw new Error(`${this.name} requires metadata and searchForms to work on.`);
```

Comment:
This error message can be improved. Something like this would be good: https://github.com/comunica/comunica/blob/master/packages/actor-rdf-resolve-quad-pattern-qpf/lib/ActorRdfResolveQuadPatternQpf.ts#L53-L54

Reply:
Actually, never mind, such an error is thrown in another actor.